README-relationships.md

```diff
@@ -0,0 +1,262 @@
+
+```ts
+enum CodeGenConnectionType {
+  HAS_ONE = 'HAS_ONE',
+  BELONGS_TO = 'BELONGS_TO',
+  HAS_MANY = 'HAS_MANY',
+}
+
+type CodeGenConnectionTypeBase = {
+  kind: CodeGenConnectionType;
+  connectedModel: CodeGenModel;
+    // ^-- Type not shown
+};
+
+type CodeGenFieldConnectionBelongsTo = CodeGenConnectionTypeBase & {
+  kind: CodeGenConnectionType.BELONGS_TO;
+  targetNames: string[];
+}
+
+type CodeGenFieldConnectionHasOne = CodeGenConnectionTypeBase & {
+  kind: CodeGenConnectionType.HAS_ONE;
+  associatedWith: CodeGenField[];
+    // ^-- Type not shown -- rendered in MIS as a string array
+  targetNames: string[];
+}
+
+export type CodeGenFieldConnectionHasMany = CodeGenConnectionTypeBase & {
+  kind: CodeGenConnectionType.HAS_MANY;
+  associatedWith: CodeGenField[];
+    // ^-- Type not shown -- rendered in MIS as a string array
+}
+```
+
+Considering a snippet of the above sample:
+
+```json
+  "models": {
+    "Primary": {
+      "name": "Primary",
+      "fields": {
+        "relatedMany": {
+          "name": "relatedMany",
+          "isArray": true,
+          "type": {
+            "model": "RelatedMany"
+          },
+          "isRequired": false,
+          "attributes": [],
+          "isArrayNullable": true,
+          "association": {
+            "connectionType": "HAS_MANY",
+            "associatedWith": ["primaryId"]
+          }
+        },
+...
+    "RelatedMany": {
+      "name": "RelatedMany",
+      "fields": {
+        "primary": {
+          "name": "primary",
+          "isArray": false,
+          "type": {
+            "model": "Primary"
+          },
+          "isRequired": false,
+          "attributes": [],
+          "association": {
+            "connectionType": "BELONGS_TO",
+            "targetNames": ["primaryId"]
+          }
+        },
+        "primaryId": {
+          "name": "primaryId",
+          "isArray": false,
+          "type": "ID",
+          "isRequired": false,
+          "attributes": []
+        }
+```
+
+- `models.Primary` - A type definition. The **source type** for any `association`s defined in this model.
+- `models.Primary.fields.relatedMany` - The **association field**/**connection field**
+- `models.Primary.fields.relatedMany.type` - The **associated type** for this relationship. This must be a `@model`.
+- `models.Primary.fields.relatedMany.association` - The structure containing the data needed to navigate the relationship with the associated type
+- `models.Primary.fields.relatedMany.association.connectionType` - The kind of relationship (has one, has many, belongs to) this **source type** has with the associated type
+- `models.Primary.fields.relatedMany.association.associatedWith` - A list of fields on the **associated type** that hold the primary key of the **source** record. This is an array so we can support composite primary keys.
+- `models.RelatedOne` - A type definition. The **source type** for any `association`s defined in this model.
```

Comment:
In the `Considering a snippet...` section the `models.RelatedMany` is attached but the explanation uses the `models.RelatedOne`

Reply:
D'oh! Thanks for catching. Fixed on 008fa1e